agent/whois_domain_agent.py

```diff
@@ -50,15 +50,17 @@ def process(self, message: msg.Message) -> None:
         if domain is None:
             return
 
-        logger.info("Processing message of selector : %s", message.selector)
-        if self.set_add("agent_whois_domain_asset", domain) is False:
-            logger.info("target %s was processed before, exiting", domain)
+        domain_object = tld.get_tld(domain, as_object=True, fix_protocol=True)
+
+        logger.info("Processing message of selector : %s.", message.selector)
+        if self.set_add("agent_whois_domain_asset", domain_object.fld) is False:
+            logger.info("target %s was processed before, exiting", domain_object.fld)
             return
-        if self._is_domain_in_scope(domain) is False:
+        if self._is_domain_in_scope(domain_object.fld) is False:
```

Comment:
I don't understand the fix here, if it is already domain why domain_object.fld??

Reply:
domain is for instance www.google.com, fld would be google.com.